validator/db/manage_test.go

```diff
@@ -0,0 +1,214 @@
+package db
+
+import (
+	"bytes"
+	"context"
+	"os"
+	"testing"
+
+	"github.com/pkg/errors"
+	"github.com/prysmaticlabs/go-bitfield"
+	slashpb "github.com/prysmaticlabs/prysm/proto/slashing"
+	"github.com/prysmaticlabs/prysm/shared/testutil"
+)
+
+type sourceStoresHistory struct {
+	ProposalEpoch                       uint64
+	FirstStoreFirstPubKeyProposals      bitfield.Bitlist
+	FirstStoreSecondPubKeyProposals     bitfield.Bitlist
+	SecondStoreFirstPubKeyProposals     bitfield.Bitlist
+	SecondStoreSecondPubKeyProposals    bitfield.Bitlist
+	FirstStoreFirstPubKeyAttestations   map[uint64]uint64
+	FirstStoreSecondPubKeyAttestations  map[uint64]uint64
+	SecondStoreFirstPubKeyAttestations  map[uint64]uint64
+	SecondStoreSecondPubKeyAttestations map[uint64]uint64
+}
+
+func TestMerge(t *testing.T) {
```

Comment:
I moved the main test over to the `db` package. The two other tests have to stay at the account level because they concern empty directories, and this logic lives inside accounts. I made the account-level tests simpler.